Source/Core/Core/HW/WiiSave.cpp

```diff
@@ -41,525 +48,509 @@ constexpr Md5 s_md5_blanker{{0x0E, 0x65, 0x37, 0x81, 0x99, 0xBE, 0x45, 0x17, 0xA
-    ERROR_LOG(CONSOLE, "Cannot open %s", m_encrypted_save_path.c_str());
-    m_valid = false;
-    return;
+    File::IOFile banner_file(m_wii_title_path + "/banner.bin", "wb");
+    return banner_file.WriteBytes(header.banner, header.hdr.banner_size);
   }
 
-  data_file.Seek(HEADER_SZ + BK_SZ, SEEK_SET);
+  bool WriteBkHeader(const BkHeader& bk_header) override { return true; }
 
-  FileHDR file_hdr_tmp;
-
-  for (u32 i = 0; i < m_bk_hdr.number_of_files; ++i)
+  bool WriteFiles(const std::vector<SaveFile>& files) override
   {
-    memset(&file_hdr_tmp, 0, FILE_HDR_SZ);
-    m_iv.fill(0);
-
-    if (!data_file.ReadBytes(&file_hdr_tmp, FILE_HDR_SZ))
-    {
-      ERROR_LOG(CONSOLE, "Failed to read header for file %d", i);
-      m_valid = false;
-    }
-
-    if (file_hdr_tmp.magic != FILE_HDR_MAGIC)
-    {
-      ERROR_LOG(CONSOLE, "Bad File Header");
-      break;
-    }
-    else
+    for (const SaveFile& file : files)
     {
       // Allows files in subfolders to be escaped properly (ex: "nocopy/data00")
       // Special characters in path components will be escaped such as /../
-      std::string file_path = Common::EscapePath(file_hdr_tmp.name.data());
-
+      std::string file_path = Common::EscapePath(file.path);
       std::string file_path_full = m_wii_title_path + '/' + file_path;
       File::CreateFullPath(file_path_full);
-      const File::FileInfo file_info(file_path_full);
-      if (file_hdr_tmp.type == 1)
-      {
-        u32 file_size_rounded = Common::AlignUp<u32>(file_hdr_tmp.size, BLOCK_SZ);
-        std::vector<u8> file_data(file_size_rounded);
-        std::vector<u8> file_data_enc(file_size_rounded);
 
-        if (!data_file.ReadBytes(file_data_enc.data(), file_size_rounded))
-        {
-          ERROR_LOG(CONSOLE, "Failed to read data from file %d", i);
-          m_valid = false;
-          break;
-        }
-
-        m_iv = file_hdr_tmp.iv;
-        m_ios.GetIOSC().Decrypt(IOS::HLE::IOSC::HANDLE_SD_KEY, m_iv.data(), file_data_enc.data(),
-                                file_size_rounded, file_data.data(), IOS::PID_ES);
+      if (file.type == 1)
+      {
+        File::IOFile raw_save_file(file_path_full, "wb");
+        const std::optional<std::vector<u8>>& data = *file.data;
+        if (!data)
+          return false;
+        raw_save_file.WriteBytes(data->data(), data->size());
+      }
+      else if (file.type == 2)
+      {
+        File::CreateDir(file_path_full);
+        if (!File::IsDirectory(file_path_full))
+          return false;
+      }
+    }
+    return true;
+  }
 
-        INFO_LOG(CONSOLE, "Creating file %s", file_path_full.c_str());
+private:
+  void ScanForFiles()
+  {
+    std::vector<std::string> directories;
+    directories.push_back(m_wii_title_path);
+    u32 size = 0;
 
-        File::IOFile raw_save_file(file_path_full, "wb");
-        raw_save_file.WriteBytes(file_data.data(), file_hdr_tmp.size);
+    for (u32 i = 0; i < directories.size(); ++i)
+    {
+      if (i != 0)
```

Comment:
Why is the first one skipped? Because it is `.` or `..`? If so, can you change the check to be more explicit for that instead?